src/Sentry/SentrySdk.cs

```diff
@@ -17,8 +22,27 @@ public static partial class SentrySdk
 {
     internal static IHub CurrentHub = DisabledHub.Instance;
 
+    /// <summary>
+    /// The Build Variables generated from you csproj file and initialized by the Sentry Source Generated Module Initializer
+    /// </summary>
+    public static IReadOnlyDictionary<string, string>? BuildVariables { get; private set; }
+
     internal static SentryOptions? CurrentOptions => CurrentHub.GetSentryOptions();
 
+
+    /// <summary>
+    /// This is called by a Sentry Source-Generator module initializers to help us determine things like
+    ///     Is your app AOT
+    /// </summary>
+    /// <param name="variables"></param>
+    public static void InitializeBuildVariables(Dictionary<string, string> variables)
```

Comment:
question: Should we "bury" the new public property and method in a new type that we "hide" a little bit?

Indeed - technically - user code cannot do too much "harm" here since it's likely that `AotHelper.IsTrimmed` is already evaluated and cached, even when user code calls this method.
Yet, I am wondering if we should move this new method and property to a different type, which is `public`, but in a more "hidden" namespace.

E.g. `Sentry.CompilerServices.BuildHelpers`

To avoid it being "accidentally discovered" by user code. And also to slightly declutter the `SentrySdk` type.
What do you think?

Reply:
DOne